### PR TITLE
Resolve pipeline datasets matching patterns defined in the catalog

### PR DESCRIPTION
**Pull request by pierre-godard**
## Description

When a tracking dataset is defined in the pipeline outputs, but not directly in the data catalog and if it matches a data set pattern defined in the data catalog, then this dataset won't be detected by kedro.

This PR fixes this behavior by adding a dataset resolution step after the catalog and pipelines have been loaded.

## Development notes

* The `demo-project` has been updated to use dataset patterns in the tracking layer.

* Use of the "resolve" term as this is similar to the resolve command mentioned here: https://github.com/kedro-org/kedro/issues/2603

## Checklist

- [x] Read the [contributing](/CONTRIBUTING.md) guidelines
- [x] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Updated the documentation to reflect the code changes
- [x] Added new entries to the `RELEASE.md` file
- [ ] Added tests to cover my changes

<a href="https://gitpod.io/#https://github.com/kedro-org/kedro-viz/pull/1491"></a>

### CODE REVIEW

File: package/kedro_viz/server.py [hunk 1]
```diff
@@ -27,6 +34,29 @@ def is_localhost(host) -> bool:
...
+    # Sort data sets by name, then by namespace to display similar data sets together
+    sorted_data_set_names = sorted(
+        data_set_names, key=lambda name: ".".join(reversed(name.split(".")))
+    )
+
+    for data_set_name in sorted_data_set_names:
+        try:
+            catalog._get_dataset(data_set_name)  # pylint: disable=protected-access
+        except DatasetNotFoundError:
+            continue
```
[the remark this answers is not in the document]
Comment (ankatiyar) [reply package/kedro_viz/server.py:57]: This internally calls `catalog._get_dataset(dataset)` and registers it to the catalog so subsequent catalog.list() calls will have the factory datasets too

File: package/kedro_viz/server.py [hunk 2]
```diff
@@ -27,6 +34,29 @@ def is_localhost(host) -> bool:
...
 
+def resolve_data_set_patterns(catalog: DataCatalog, pipelines: Dict[str, Pipeline]):
+    """Resolve data set patterns in the data catalog by matching them against
+    the data sets in the pipelines.
+    """
+
+    data_set_names = {
+        data_set_name
+        for pipeline in pipelines.values()
+        for data_set_name in pipeline.data_sets()
```
Comment (ankatiyar) [inline package/kedro_viz/server.py:45]: ALso just flagging to the team that this method is renamed to `pipeline.datasets()` on the Kedro `develop` branch

**Comment by ravi-kumar-pilla** [comment]
Hi @pierre-godard , Thank you for the PR. WIll it be possible for you to discard the files being renamed from the PR and make changes as suggested by @ankatiyar ? For your reference, I followed your approach and created a draft [here](https://github.com/kedro-org/kedro-viz/pull/1588).

Thank you

**Comment by ravi-kumar-pilla** [comment]
Closing this in favor of #1588

**Pull request closed without merging**